sql/data_sql_warehouse.go

```diff
@@ -31,11 +32,36 @@ func DataSourceWarehouse() *schema.Resource {
 
 	return common.DataResource(SQLWarehouseInfo{}, func(ctx context.Context, e interface{}, c *common.DatabricksClient) error {
 		data := e.(*SQLWarehouseInfo)
-		err := c.Get(ctx, fmt.Sprintf("/sql/warehouses/%s", data.ID), nil, data)
+		var id string
+		endpointsAPI := NewSQLEndpointsAPI(ctx, c)
+		if data.Name != "" {
+			endpoints, err := endpointsAPI.List()
+			if err != nil {
+				return err
+			}
+			selected := []SQLEndpoint{}
+			lname := strings.ToLower(data.Name)
```

Comment:
Why not exact match? Is this a pattern we have for data sources when looking up by name today? It would fail if two warehouses have the same name modulo case, e.g. "test" and "Test".

Reply:
It's just for convenience. Our UI doesn't allow to create warehouses with the same name, but different cases. But yes, it looks like other resources aren't supporting that - jobs & clusters use exact matching.

Reply:
will remove that tomorrow.